src/main/java/tasks/Event.java

```diff
@@ -0,0 +1,57 @@
+package tasks;
+
+import java.time.LocalDateTime;
+
+import common.DateTimeParser;
+
+public class Event extends Task {
+    private String from;
+    private String to;
+    private LocalDateTime fromDate;
+    private LocalDateTime toDate;
+
+    public Event(String description, String from, String to) {
+        super(description);
+        this.from = from;
+        this.to = to;
+        try {
+            this.fromDate = DateTimeParser.parse(from);
+        } catch (IllegalArgumentException e) {
+            //No correct format, deadline only recorded as a String
+            this.fromDate = null;
+        } try {
+            this.toDate = DateTimeParser.parse(to);
+        } catch (IllegalArgumentException e) {
+            //No correct format, deadline only recorded as a String
+            this.toDate = null;
+        }
+    }
+
+    public LocalDateTime getDate() {
+        return this.toDate;
+    }
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + DateTimeParser.format(fromDate, from) + " to: " + DateTimeParser.format(toDate, to) + ")";
+    }
+
+    /**
+     * Returns an Event given a String that represents the Event task.
+     *
+     * @param str The String that represents the Event Task.
+     * @return
+     */
+    public static Event fromString(String str) {
+        boolean isDone = str.contains("[X]");
```

Comment:
What if some weird case inputs `"[X]"` as the task description? It might be better to confirm that the `"[X]"` is at the right place (i.e. from index 3 to 5) if at all present.